### PR TITLE
[WIP] new invite command

### PR DESCRIPTION
**Pull request by amurdock**

### CODE REVIEW

File: internal/routes/admin/invite/invite.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	return func(context *gin.Context) {
+		payload := new(AdminInvitePayload)
+		if err := context.BindJSON(&payload); err != nil {
+			context.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"message": err.Error()})
+			return
+		}
+
+		_, err := AdminMemberInvite(store, payload)
+		if err != nil {
+			return
```
Comment (amurdock) [inline internal/routes/admin/invite/invite.go:52]: Need to handle this error.

File: internal/store/store.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+	var payload JSONB
+	errUnmarshal := json.Unmarshal(data, &payload)
+	if errUnmarshal != nil {
+		return 0, errUnmarshal
+	}
+
+	var evt Event
+
+	result := es.db.Model(&Event{}).Where("entity_id = ?", entityId).Order("entity_seq desc").Take(&evt)
```
Comment (amurdock) [inline internal/store/store.go:56]: Needs to be within transaction.

File: internal/routes/admin/invite/invite_test.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+func assertStore(t *testing.T, mock *MockEventStore) {
+	mock.AssertExpectations(t)
+
+	args := mock.Calls[0].Arguments
+	uuid := args.String(0)
+	event := args.Get(1).(*interface{})
+
+	assert.Regexp(t, UuidV4Regex, uuid)
```
Comment (hellojakecross) [inline internal/routes/admin/invite/invite_test.go:36]: The validity of the UIUD should part of the event stores responsibilities and tested in there. As long as we're passing it along from the invite, that should be enough from here.

File: internal/routes/admin/invite/invite.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+type AdminInviteCommand struct {
+	Id    string `json:"id" binding:"required"`
+	Name  string `json:"name" binding:"required"`
+	Email string `json:"email" binding:"required"`
+}
+
+type AdminInviteCreatedEvent events.Event[AdminInviteCommand]
+
+func AdminMemberInvite(s store.EventStore, payload *AdminInvitePayload) (uint, error) {
```
Comment (hellojakecross) [inline internal/routes/admin/invite/invite.go:28]: Does payload need to be a pointer?

File: internal/store/store.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+)
+
+type JSONB map[string]interface{}
+
+func (j JSONB) Value() (driver.Value, error) {
+	valueString, err := json.Marshal(j)
+	return string(valueString), err
+}
+
+func (j *JSONB) Scan(value interface{}) error {
```
Comment (hellojakecross) [inline internal/store/store.go:19]: If we're exporting a method, we should probably have a test for it explicitly.

File: internal/store/store.go [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+	result = es.db.Create(&Event{
+		EntityID:  entityId,
+		EntitySeq: sequence,
+		Payload:   payload,
+	})
+
+	return sequence, result.Error
+}
+
+func New(db *gorm.DB) EventStore {
```
Comment (hellojakecross) [inline internal/store/store.go:75]: From what I've seen, it's idiomatic to call these types of function `NewType` so `NewEventStore` - that would mean we could have other `NewType` functions available from the same package (and makes DI registration clearer)

File: .gitignore [hunk 7]
```diff
@@ -1,3 +1,4 @@
 .env
 .idea
 bin
+tmp
```
[the remark this answers is not in the document]
Comment (amurdock) [reply .gitignore:4]: Occasionally I get a `tmp` file created. Not sure why at the moment. I'll remove until I've worked out who/what is creating this.

File: internal/store/store.go (hunk 2 above)
Comment (amurdock) [reply internal/store/store.go:56]: https://gorm.io/docs/transactions.html#Disable-Default-Transaction

File: internal/routes/admin/invite/invite_test.go (hunk 3 above)
Comment (amurdock) [reply internal/routes/admin/invite/invite_test.go]: I've moved the validation of the `EntityId` into internal/store/store.go. And removed the associated assertion.

File: internal/routes/admin/invite/invite.go (hunk 4 above)
Comment (amurdock) [reply internal/routes/admin/invite/invite.go]: Nope ... sorted 👍🏻

File: internal/store/store.go (hunk 6 above)
Comment (amurdock) [reply internal/store/store.go]: Now called `NewEventStore`.

**Pull request merged**